go-controller/pkg/ovn/gateway.go

```diff
@@ -721,17 +643,33 @@ func (gw *GatewayManager) GatewayInit(
 			}
 		}
 	}
+	return nil
+}
 
+// updateGWRouterStaleNAT updates the SNAT rules on the gateway router that are created outside the GatewayManager.
+// Multiple handlers, like
```

Comment:
thanks for calling this out. let's take a deeper look into this after this PR goes in.